src/topology/homotopy/equiv.lean

```diff
@@ -0,0 +1,140 @@
+
+/--
+Any topological space is homotopy equivalent to itself.
+-/
+def refl (X : Type u) [topological_space X] : homotopy_equiv X X :=
+{ to_fun := id,
+  inv_fun := id,
+  left_inv := homotopic.refl _,
+  right_inv := homotopic.refl _ }
+
+instance : inhabited (homotopy_equiv unit unit) := ⟨refl unit⟩
+
+@[simp]
+lemma coe_refl (X : Type u) [topological_space X] : ⇑(refl X) = id := rfl
+
+/--
+If `X` is homotopy equivalent to `Y`, then `Y` is homotopy equivalent to `X`.
+-/
+def symm (h : homotopy_equiv X Y) : homotopy_equiv Y X :=
+{ to_fun := h.inv_fun,
+  inv_fun := h.to_fun,
+  left_inv := h.right_inv,
+  right_inv := h.left_inv }
+
+@[simp]
+lemma coe_inv_fun (h : homotopy_equiv X Y) : (⇑h.inv_fun : Y → X) = ⇑h.symm := rfl
+
+@[simp]
+lemma refl_symm (X : Type u) [topological_space X] : (refl X).symm = refl X := rfl
+
+/--
+If `X` is homotopy equivalent to `Y`, and `Y` is homotopy equivalent to `Z`, then `X` is homotopy
+equivalent to `Z`.
+-/
+def trans (h₁ : homotopy_equiv X Y) (h₂ : homotopy_equiv Y Z) : homotopy_equiv X Z :=
+{ to_fun := h₂.to_fun.comp h₁.to_fun,
+  inv_fun := h₁.inv_fun.comp h₂.inv_fun,
+  left_inv := begin
+    refine homotopic.trans _ h₁.left_inv,
+    change ((h₁.inv_fun.comp h₂.inv_fun).comp (h₂.to_fun.comp h₁.to_fun)) with
+      h₁.inv_fun.comp ((h₂.inv_fun.comp h₂.to_fun).comp h₁.to_fun),
+    refine homotopic.hcomp _ (homotopic.refl _),
+    refine homotopic.trans ((homotopic.refl _).hcomp h₂.left_inv) _,
+    rw id_comp,
+  end,
+  right_inv := begin
+    refine homotopic.trans _ h₂.right_inv,
+    change ((h₂.to_fun.comp h₁.to_fun).comp (h₁.inv_fun.comp h₂.inv_fun)) with
+      h₂.to_fun.comp ((h₁.to_fun.comp h₁.inv_fun).comp h₂.inv_fun),
+    refine homotopic.hcomp _ (homotopic.refl _),
+    refine homotopic.trans ((homotopic.refl _).hcomp h₁.right_inv) _,
+    rw id_comp,
+  end }
+
+@[simp]
+lemma coe_trans (h₁ : homotopy_equiv X Y) (h₂ : homotopy_equiv Y Z) :
+  ⇑(h₁.trans h₂) = h₂ ∘ h₁ := rfl
+
+lemma symm_trans (h₁ : homotopy_equiv X Y) (h₂ : homotopy_equiv Y Z) :
+  (h₁.trans h₂).symm = h₂.symm.trans h₁.symm := by ext; refl
+
+end homotopy_equiv
+
+end continuous_map
+
+open continuous_map
+
+namespace homeomorph
+
+/--
+Any homeomorphism is a homotopy equivalence.
+-/
+def to_homotopy_equiv (h : X ≃ₜ Y) : homotopy_equiv X Y :=
+{ to_fun := ⟨h⟩,
+  inv_fun := ⟨h.symm⟩,
+  left_inv := by { convert homotopic.refl _, ext, simp },
+  right_inv := by { convert homotopic.refl _, ext, simp } }
+
+@[simp]
+lemma coe_to_homotopy_equiv (h : X ≃ₜ Y) : ⇑(h.to_homotopy_equiv) = h := rfl
+
+@[simp]
+lemma refl_to_homotopy_equiv (X : Type u) [topological_space X] :
+  (homeomorph.refl X).to_homotopy_equiv = homotopy_equiv.refl X := rfl
+
+@[simp]
+lemma symm_to_homotopy_equiv (h : X ≃ₜ Y) :
+  h.symm.to_homotopy_equiv = h.to_homotopy_equiv.symm := rfl
+
+@[simp]
+lemma trans_to_homotopy_equiv (h₀ : X ≃ₜ Y) (h₁ : Y ≃ₜ Z) :
+  (h₀.trans h₁).to_homotopy_equiv = h₀.to_homotopy_equiv.trans h₁.to_homotopy_equiv := rfl
```

Comment:
I'm not 100% sure which way round these two should go.

Reply:
My intuition says the other way round: `.to_homotopy_equiv` "throws away information" so that should maybe be the last thing you do.

Reply:
See [this thread](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/simp.20normal.20form.20for.20.60e.2Eto_equiv.2Esymm.60) for a related discussion.

Reply:
I've had a look at the thread but I'm not sure a conclusion was reached?

Reply:
I suggest that we leave it this way. E.g., one can have simp lemmas for `h₀.to_homotopy_equiv`.